src/main/kotlin/lotto/LottoMachine.kt

```diff
@@ -0,0 +1,34 @@
+package lotto
+
+class LottoMachine(val purchaseAmount: Int, val tickets: MutableList<Lotto> = emptyList<Lotto>().toMutableList()) {
+    private val numberList = (Lotto.Companion.MIN..Lotto.Companion.MAX)
+    private var change = 0
+    private var ticketCount = 0
+
+    init {
+        require(purchaseAmount in MIN..MAX) {
+            throw IllegalArgumentException()
+        }
```

Comment:
constructor based validation! 👍
the `require` already include throw exception 😄

<img width="512" alt="image" src="https://github.com/user-attachments/assets/1946b239-0638-4064-820d-b3592fdf86e8" />

and how about adding a message to help users quickly decide their next action?

```suggestion
    init {
        require(purchaseAmount in MIN..MAX) { "blah blah blah" }
```